### PR TITLE
refactor(Gatsby): Upgrade npm dependencies

### PR DESCRIPTION
**Pull request by Vesely**

### CODE REVIEW

**Comment by rostaklein** [comment]
Most of the lint fails because of updated eslint I assume? 🤔

File: gatsby/src/components/footer/footer.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ const DRUPAL_INTERNAL_IDS = {
 };
 
 const Footer: React.FC = () => {
-  const result = useStaticQuery<IFooterLinksQuery>(query);
+  const result = useStaticQuery<IFooterLinksQuery>(query); // eslint-disable-line @typescript-eslint/no-use-before-define
```
Comment (rostaklein) [inline gatsby/src/components/footer/footer.tsx:18]: Could you please just move the query inside of this function? :)

File: gatsby/src/components/button/button.tsx [hunk 2]
```diff
@@ -41,6 +41,7 @@ type AnchorProps = React.AnchorHTMLAttributes<HTMLAnchorElement> &
   IButtonBase & { href: string };
 
 // Input/output options
+/* global JSX */
```
Comment (rostaklein) [inline gatsby/src/components/button/button.tsx:44]: Whats wrong with this? 🤔

Comment (Vesely) [reply gatsby/src/components/button/button.tsx:44]: @rostaklein without that eslint throws an error `"'JSX' is not defined"  no-undef`

**Comment by rostaklein** [comment]
Seems all good to me, lets merge this 🚀

**Review by rostaklein — APPROVED** [review]

**Pull request merged**